### PR TITLE
Stroke medicine review

### PR DESCRIPTION
**Pull request by medic-code**
I've reviewed the stroke medicine exercises see the comments below. 

# Changes made 
1. Minor grammatical changes 
2. Added some code explanations in part 2 for the plots
3. Added explanation for NaN values in part3
4. Added some interactivity around selecting data in part 3

# General feedback

## Overall comments

I enjoyed doing this exercise, a lot to learn about preparing data and what feels like more life like way of looking a dataset and trying out different models. There's some really great content here that dives deeper into some of the problems with datasets and what to do about this. Definitely, an extension of the breast cancer dataset,  expanding on the data preprocessing side. There are parts to the exercise I think need a bit of expansion for a student who have not come across some of the concepts. This exercise covers a lot of ground, though.

It may be worth encouraging the students to look at the docs of these packages to understand the syntax around some of the methods we use so they get into that habit.

## Learn experience

I learned a lot from this exercise about data processing and how to do something about it. We introduce a few different packages here, which is inevitable when working with a dataset. I think there are parts that, without some expansion, may give the students a little difficulty. This is perhaps not bad as it encourages them to look things up. 

I think there's enough in the exercise tasks-wise while still introducing a lot of new concepts for the students to take something away yet get their hands dirty.

# Specific feedback

## Part 1 
1. To consider consistency in the text for setting up the exercises. I notice that with different authors, each one will have different ways of writing about how to set up the exercise. I liked the breast cancer text's explanation, so consider whether we use a baseline version with small additions related to the specific dataset.

## Part 3

1. In the preprocessing data section, we mention impute. I wonder if we add some info on this, as it's probably the first time they've heard this term.

2. We use `random_state` in the `train_test_split` method - It might be worth mentioning the reason for using this parameter for this method.

3. In the categorical sub-section of data preprocessing, we specify named, ordinal, and nominal. I wonder if we add more info to what those mean, as in the sections below; this assumes some knowledge.

4. In the 'nominal categories' part of the categorial sub-section, we mention not wanting to label categories as a rank. I wonder if we just explain this term as it's the first time it's being used.

5. In the nominal categories, we mention One Hot encoding. I wonder, briefly expanding on what that means ,as it's the first time we see this term. The table below does suggest what it is, but it may be useful to be explicit.

6. In the 'using data pipelines' section - I wonder if it's worth expanding a little bit on what a pipeline means, as it refers to methods and transformations but not necessarily explaining what those methods/transformations are. Also may be worth just going over what the code is doing a little bit more, as this is the first time they'll be exposed to the idea of a pipeline.

## Part 4

1. We mention logistic regression. I wonder if we should add a few more details as to why this model is a good starting point and the context of the model (we mention some alternatives in the next steps). This may be the first time they'll be thinking about this type of problem and so worth being explicit about model choice.

2. We're using the accuracy score method; worth expanding a bit on what this is so students understand that it's a metric to evaluate logistic regression.

3. In the section about F1 - we mention harmonic mean; it may be worth defining this term as it'll probably be the first time a student will have come across this.

4. At the end of the F1 section, we're using Support Vector Machine, Random Forest, and Gradient Boosting models; worth expanding a sentence on why we're using these models. Also, explain a little bit about the code methods below the code snippet using these models.

## Part 5

1. We use the `class_weight="balanced" argument in the logistic regression method. We explain the parameter in the text but not necessarily what the argument 'balanced' means.

2. Consider adding a small sentence on the imblearn package as it's our first introduction. In the code snippet itself is worth expanding on the meaning of `sampling strategy` parameter and its argument.

3. Similarly, in part 4, I wonder if we need to provide some context to the other models used. A lot of them our repeated from part 4 so might be overkill if we've already explained about them in part4. Additionally is there a reason we're using KNN for example here but not in part 4 ? Reason I mention this is that its another model we haven't introduced in the exercise.

4. At the end of this section, we find out the model's accuracy is poor. I wonder if we need to add a little more to what to do with this result. It might be that let's follow the next steps or accept that, given this dataset, that's the best we're going to get. 

## Additional interactivity 

1. In the model penalisation where we write code to do this, getting the students to do a class weight for different models (they've already been introduced to the concept of testing the dataset on new models and have an example of how to do this on logistic regression.

### CODE REVIEW

**Comment by chris-lovejoy** [comment]
Thanks so much for the thorough review Aaron - really, really helpful. I've implemented all the changes now based on your feedback - closing the PR
